### PR TITLE
feat(ecs): add data-sources availability zones and compute flavors.

### PR DESCRIPTION
**Pull request by baixinsui**
**What this PR does / why we need it**:
This PR is used to add data-sources availability zones and compute flavors .
**Special notes for your reviewer**:

**Release note**:

```release-note
add data-sources availability zones and compute flavors
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./internal' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-hcso/internal       70.796s
```

### CODE REVIEW

File: docs/data-sources/compute_flavors.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+* `performance_type` - (Optional, String) Specifies the ECS flavor type. Possible values are as follows:
+  + **normal**: General computing
+  + **computingv3**: General computing-plus
+  + **highmem**: Memory-optimized
+  + **saphana**: Large-memory HANA ECS
+  + **diskintensive**: Disk-intensive
+
+* `generation` - (Optional, String) Specifies the generation of an ECS type. For example, **s3** indicates
+  the general-purpose third-generation ECSs. For details, see
+  [ECS Specifications](https://support.hcso.com/intl/en-us/productdesc-ecs/ecs_01_0014.html).
```
Comment (iskey) [inline docs/data-sources/compute_flavors.md]: This link is not available, Please remove it.

File: docs/json/data-sources/compute_flavors.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
+  "provider_schemas": {
```
Comment (iskey) [inline docs/json/data-sources/compute_flavors.json]: This file is not needed.

File: internal/data_source_huaweicloud_availability_zones.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package internal
+
+import (
+	"sort"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/openstack/compute/v2/extensions/availabilityzones"
```
Comment (iskey) [inline internal/data_source_huaweicloud_availability_zones.go]: should have a break line here for different packages.

File: internal/services/ecs/data_source_huaweicloud_compute_flavors.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
+package ecs
```
Comment (iskey) [inline internal/services/ecs/data_source_huaweicloud_compute_flavors.go]: Try to import from the Huawei cloud but not implementing it here.

**Review by iskey — CHANGES_REQUESTED** [review]
comments added

File: docs/json/data-sources/compute_flavors.json (hunk 2 above)
Comment (baixinsui) [reply docs/json/data-sources/compute_flavors.json]: Removed

File: internal/data_source_huaweicloud_availability_zones.go (hunk 3 above)
Comment (baixinsui) [reply internal/data_source_huaweicloud_availability_zones.go]: Removed it and import it from `huaweicloud`

File: internal/services/ecs/data_source_huaweicloud_compute_flavors.go (hunk 4 above)
Comment (baixinsui) [reply internal/services/ecs/data_source_huaweicloud_compute_flavors.go]: Ok

File: docs/data-sources/compute_flavors.md (hunk 1 above)
Comment (baixinsui) [reply docs/data-sources/compute_flavors.md]: Removed

**Review by iskey — APPROVED** [review]
LGTM

**Pull request merged**